ocs_ci/ocs/replica_one.py

```diff
@@ -0,0 +1,199 @@
+from logging import getLogger
+
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pod import (
+    get_pods_having_label,
+    get_ceph_tools_pod,
+    run_osd_removal_job,
+)
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.constants import (
+    DEFAULT_CEPHBLOCKPOOL,
+    OSD_APP_LABEL,
+    CEPHBLOCKPOOL,
+    STORAGECLASS,
+    DEPLOYMENT,
+)
+from ocs_ci.ocs.exceptions import CommandFailed
+
+
+log = getLogger(__name__)
+
+REPLICA1_STORAGECLASS = "ocs-storagecluster-ceph-non-resilient-rbd"
+
+# For local testing only. to be removed
+config.ENV_DATA["worker_availability_zones"] = [
+    "us-south-1",
+    "us-south-2",
+    "us-south-3",
+]
+# End
+
+
+def get_failures_domain_name() -> list[str]:
+    """
+    Fetch Failure domains from cephblockpools names
+
+    Returns:
+        list[str]: list with failure domain names
+
+    """
+    cpb_object = OCP(kind=CEPHBLOCKPOOL)
+    failure_domains = list()
+    cephblockpools_names = list()
+    prefix = DEFAULT_CEPHBLOCKPOOL
+    for i in range(0, len((cpb_object.data["items"]))):
+        cephblockpools_names.append(cpb_object.data["items"][i]["metadata"]["name"])
+
+    log.info(f"Cephblockpool names:{cephblockpools_names}")
+
+    for name in cephblockpools_names:
+        if name.startswith(prefix):
+            corrected_name = name[len(prefix) :].lstrip("-")
+            log.info(corrected_name)
+            if corrected_name:
+                failure_domains.append(corrected_name)
+
+    log.info(f"Failure domains:{failure_domains}")
+
+    return failure_domains
+
+
+def get_replica_1_osds() -> tuple[list[str], list[str]]:
+    """
+    Gets the names and IDs of OSD associated with replica1
+
+    Returns:
+        tuple[list[str], list[str]]: list with OSDs names, list with OSD IDs
+
+    """
+    replica1_osds_id = list()
+    replica1_osds = list()
+    all_osds = get_pods_having_label(label=OSD_APP_LABEL)
+    for domain in config.ENV_DATA["worker_availability_zones"]:
+        for osd in all_osds:
+            if osd["metadata"]["labels"]["ceph.rook.io/DeviceSet"] == domain:
+                replica1_osds.append(osd["metadata"]["name"])
+                replica1_osds_id.append(osd["metadata"]["labels"]["ceph-osd-id"])
+    log.info(replica1_osds)
+    return (replica1_osds, replica1_osds_id)
```

Comment:
I think it will be safer to return a dictionary of OSD name and the matching OSD ID. Consider applying this change:

```python
    replica1_osds = dict()
    for domain in config.ENV_DATA["worker_availability_zones"]:
        for osd in all_osds:
            if osd["metadata"]["labels"]["ceph.rook.io/DeviceSet"] == domain:
                replica1_osds.update({"osd_name": osd["metadata"]["name"], "osd_id": osd["metadata"]["labels"]["ceph-osd-id"]})
    log.info(replica1_osds)
    return replica1_osds
```